apiserver/facades/client/bundle/bundle.go

```diff
@@ -151,10 +156,111 @@ func (b *BundleAPI) GetChanges(args params.BundleChangesParams) (params.BundleCh
+	exportConfig.SkipCloudImageMetadata = true
+	exportConfig.SkipCredentials = true
+	exportConfig.SkipIPAddresses = true
+	exportConfig.SkipSSHHostKeys = true
+	exportConfig.SkipStatusHistory = true
+	exportConfig.SkipLinkLayerDevices = true
+
+	model, err := b.backend.ExportPartial(exportConfig)
+	if err != nil {
+		return params.BytesResult{}, common.ServerError(err)
+	}
+
+	err = model.Validate()
+	if err != nil {
+		return params.BytesResult{}, common.ServerError(err)
+	}
+
+	// Fill it in charm.BundleData datastructure.
+	bundleData, err := b.FillBundleData(model)
+	if err != nil {
+		return params.BytesResult{}, common.ServerError(err)
+	}
+
+	bytes, err := yaml.Marshal(bundleData)
+	if err != nil {
+		return params.BytesResult{}, common.ServerError(err)
+	}
+
+	return params.BytesResult{
+		Result: bytes,
+	}, nil
 }
 
 // Mask the new method from V1 API.
 // ExportBundle is not in V1 API.
 func (u *APIv1) ExportBundle() (_, _ struct{}) { return }
+
+// Fills the bundledata datastructure required for the exportBundle.
+func (b *BundleAPI) FillBundleData(model description.Model) (*charm.BundleData, error) {
+	data := &charm.BundleData{
+		Series:       series.LatestLts(),
+		Applications: make(map[string]*charm.ApplicationSpec),
+		Relations:    [][]string{},
+		Machines:     make(map[string]*charm.MachineSpec),
+	}
+
+	if len(model.Applications()) == 0 {
+		return &charm.BundleData{}, errors.NotFoundf("application")
+	}
+	for _, application := range model.Applications() {
+		ut := []string{}
+		for _, unit := range application.Units() {
+			ut = append(ut, unit.Machine().Id())
+		}
+
+		newApplication := &charm.ApplicationSpec{
+			Charm:            application.CharmURL(),
+			Series:           application.Series(),
+			NumUnits:         len(application.Units()),
+			To:               ut,
+			Expose:           application.Exposed(),
+			Options:          application.CharmConfig(),
+			Annotations:      application.Annotations(),
+			EndpointBindings: application.EndpointBindings(),
+		}
+
+		data.Applications[application.Name()] = newApplication
+	}
+
+	for _, machine := range model.Machines() {
+		constraints := []string{"arch=" + machine.Constraints().Architecture(),
+			"cpu-cores=" + strconv.Itoa(int(machine.Constraints().CpuCores())),
+			"cpu-power=" + strconv.Itoa(int(machine.Constraints().CpuPower())),
+			"mem=" + strconv.Itoa(int(machine.Constraints().Memory())),
+			"root-disk=" + strconv.Itoa(int(machine.Constraints().RootDisk()))}
+
+		newMachine := &charm.MachineSpec{
+			Constraints: strings.Join(constraints, " "),
+			Annotations: machine.Annotations(),
+			Series:      machine.Series(),
+		}
+		data.Machines[machine.Id()] = newMachine
+	}
+
+	for _, relation := range model.Relations() {
+		endpointRelation := []string{}
+		for _, endpoint := range relation.Endpoints() {
+			// skipping the 'peer' role which is not of concern in exporting the current model configuration.
+			if endpoint.Role() != "peer" {
```

Comment:
You could simplify this loop a little by saying 

```go
if endpoint.Role() == "peer" {
    continue 
}```

Reply:
done